genie-web/src/main/java/com/netflix/genie/web/controllers/GenieExceptionMapper.java

```diff
@@ -137,4 +141,21 @@ public void handleConstraintViolation(
         log.error(cve.getLocalizedMessage(), cve);
         response.sendError(HttpStatus.PRECONDITION_FAILED.value(), builder.toString());
     }
+
+    /**
+     * Handle MethodArgumentNotValid  exceptions.
+     *
+     * @param response The HTTP response
+     * @param e        The exception to handle
+     * @throws IOException on error in sending error
+     */
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public void handleMethodArgumentNotValidException(
+        final HttpServletResponse response,
+        final MethodArgumentNotValidException e
+    ) throws IOException {
+        this.methodArgumentNotValidRate.increment();
+        log.error(e.getMessage(), e);
+        response.sendError(HttpStatus.PRECONDITION_FAILED.value(), e.getMessage());
```

Comment:
Isn't a 400 / Bad request more appropriate for this?
Precondition failed has to do with request headers asking for some condition that cannot be met according to https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html